src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -629,8 +629,27 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             return ret;
         }
 
-        private static Encoding GetEncoding(int codePage)
+        private static Encoding GetConsoleEncodingOrThrow(ConsoleStreamDirection encodingType)
         {
+            int codePage = (encodingType == ConsoleStreamDirection.Input) ?
+                    (int)Interop.Kernel32.GetConsoleCP() :
+                    (int)Interop.Kernel32.GetConsoleOutputCP();
+
+            int error = Marshal.GetLastWin32Error();
+            if (error == Interop.Errors.ERROR_INVALID_HANDLE)
+            {
+                throw new PlatformNotSupportedException(SR.ConsoleEncodingNotSupported);
```

Comment:
Per our other discussions, and further thinking, this case should also return the default (0). It is quite plausible to have CreateProcess and redirection without this particular method.